lib/Ix/DBIC/Result.pm

```diff
@@ -274,43 +470,250 @@ sub ix_finalize ($class) {
+
+=method ix_create_check($ctx, \%record)
+
+A check method called before Foo/set#create operations.
+
+=method ix_update_check($ctx, $row, \%record)
+
+A check method called before Foo/set#update operations.
+
+=method ix_destroy_check($ctx, $row)
+
+A check method called before Foo/set#destroy operations.
+
+=method ix_changes_check($ctx, \%arg)
+
+A check method called before Foo/changes operations.
+
+=method ix_query_check($ctx, \%arg, \%search)
+
+A check method called before Foo/query operations. (You must define this method
+if C<ix_query_enabled> is true for your rclass.)
+
+=method ix_query_changes_check($ctx, \%arg, \%search)
+
+A check method called before Foo/queryChanges operations. (You must define this method
+if C<ix_query_enabled> is true for your rclass.)
+
+=cut
+
 sub ix_set_check { return; } # ($self, $ctx, \%arg)
 
 sub ix_get_check              { } # ($self, $ctx, \%arg)
 sub ix_create_check           { } # ($self, $ctx, \%rec)
 sub ix_update_check           { } # ($self, $ctx, $row, \%rec)
 sub ix_destroy_check          { } # ($self, $ctx, $row)
-sub ix_get_updates_check      { } # ($self, $ctx, \%arg)
+sub ix_changes_check          { } # ($self, $ctx, \%arg)
+
+=method ix_create_error($ctx, \%error)
+
+A hook method called when Foo/set#create encounters errors.
+
+=method ix_update_error($ctx, \%error)
+
+A hook method called when Foo/set#update encounters errors.
+
+=cut
 
 sub ix_create_error  { return; } # ($self, $ctx, \%error)
 sub ix_update_error  { return; } # ($self, $ctx, \%error)
 
+=method ix_created($ctx, $row)
+
+A hook method called inside the Foo/set#create transaction.
+
+=method ix_updated($ctx, $row, \%changes)
+
+A hook method called inside the Foo/set#update transaction. C<$changes> is a
+hashref with both C<old> and C<new> keys, so hooks can inspect the changes and
+take action only in certain cases.
+
+=method ix_destroyed($ctx, $row)
+
+A hook method called inside the Foo/set#destroy transaction.
+
+=cut
+
 sub ix_created   { } # ($self, $ctx, $row)
 sub ix_destroyed { } # ($self, $ctx, $row)
 
 # The input to ix_updated is not trivial to compute, so it is only called if
 # present, so we don't define it in the base class. -- rjbs, 2017-01-06
 # sub ix_updated   { } # ($self, $ctx, $row, \%changes)
 
+=method ix_postprocess_create($self, $ctx, \@rows)
+
+A postprocess hook called after Foo/set#create operations.
+
+=method ix_postprocess_update($self, $ctx, \%updated)
+
+A postprocess hook called after Foo/set#create operations.
+
+=method ix_postprocess_destroy($self, $ctx, \@row_ids)
+
+A postprocess hook called after Foo/set#create operations.
+
+=cut
+
 sub ix_postprocess_create  { } # ($self, $ctx, \@rows)
 sub ix_postprocess_update  { } # ($self, $ctx, \%updated)
 sub ix_postprocess_destroy { } # ($self, $ctx, \@row_ids)
 
+# I am not going to document this publicly, but its signature is
+# ($self, $ctx, $get_arg, $results)
 sub _return_ix_get   { return $_[3]->@* }
```

Comment:
We do explicitly state later that this method is how you'd modify results so maybe we should document it

Reply:
It's documented, -ish, when we mention it in ResultSet. I think I'd still rather leave it quiet here.